test/UserTest.cpp

```diff
@@ -119,3 +119,29 @@ TEST_F(UserTest, IsHostAllowed) {
             << "Allow-host is " << h.sMask;
     }
 }
+
+TEST_F(UserTest, TestOnlyModulesMayAuth) {
+    CUser user("user");
+    user.SetPass("password", CUser::HASH_NONE);
+
+    CZNC::Get().SetOnlyModulesMayAuth(false);
+    user.SetOnlyModulesMayAuth(false);
+
+    EXPECT_TRUE(user.CheckPass("password"));
+
+    // user-level only
+    user.SetOnlyModulesMayAuth(true);
+    EXPECT_FALSE(user.CheckPass("password"));
+
+    // re-enabling built-in authentication
+    user.SetOnlyModulesMayAuth(false);
+    EXPECT_TRUE(user.CheckPass("password"));
+
+    // on at global level, off at user level
+    CZNC::Get().SetOnlyModulesMayAuth(true);
```

Comment:
Please set this global variable to its default value in the end of test, to not accidentally affect other tests which are in the same process later.
Something like
```c++
bool bDefault = CZNC::Get().GetOnlyModules();
...
CZNC::Get().Set(bDefault);
```
Bonus points if done in an exception-safe manner.